Antidote/CallsManager.m

```diff
@@ -451,13 +465,33 @@ - (void)switchViewControllerForCall:(OCTCall *)call
 
     self.currentCallViewController = abstractVC;
 
-    self.currentCall = call;
+    self.currentCall = [RBQSafeRealmObject safeObjectFromObject:call];
 }
 
-- (BOOL)onlyPauseCallsLeft
+- (void)replaceCurrentCallThatWasRemoved
 {
-    return (([self.allActiveCallsController.fetchedObjects count] == 0) &&
-            ([self.allPausedCallsController.fetchedObjects count] != 0));
+    OCTCall *call = [self.allPausedCallsController.fetchedObjects firstObject];
+
+    if (! call) {
+        call = [self.allCallsController.fetchedObjects firstObject];
+    }
+
+    // workaround for deadlock in objcTox
+    // https://github.com/Antidote-for-Tox/objcTox/issues/51
+    [self performSelector:@selector(switchViewControllerForCall:) withObject:call afterDelay:0];
+}
+
+- (BOOL)isCurrentViewControllerReusableForCurrentCall:(OCTCall *)call
+{
+    if (! ((call.status == OCTCallStatusActive) || (call.status == OCTCallStatusPaused))) {
```

Comment:
Then there is no need for this method. We can simply check if current controller is ActiveCallViewContoller. If it is active, it means that call is active or paused.

If there is no need in this method then there will be no problem with it strange naming. :-)

Or am I wrong?